File: Medium/1493. Longest Subarray of 1's After Deleting One Element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int output = Integer.MIN_VALUE;
        List<Integer> sums = new ArrayList<>();
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                if (sum != 0) {
                    sums.add(sum);
                    sum = 0;
                }
                sums.add(0);
            } else {
                sum++;
            }
        }
        if (sum != 0) {
            sums.add(sum);
            sum = 0;
        }
        if (sums.size() == 1) {
            if (sums.get(0) == 0) {
                return 0;
            } else {
                return sums.get(0) - 1;
            }
        }
        output = sums.get(0);
        System.out.println(sums);
        for (int i = 1; i < sums.size() - 1; i++) {
            int tmp = 0;
            tmp += sums.get(i);
            tmp += sums.get(i - 1);
            tmp += sums.get(i + 1);
            output = Math.max(output, tmp);
        }
        return output;
    }
}

File: Medium/1493. Longest Subarray of 1's After Deleting One Element.py
from collections import deque
from typing import List


class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        results = []
        result = 0
        for i in range(0, len(nums) - 1):
            if nums[i] == 1:
                result += 1
            else:
                if nums[i + 1] == 1:
                    if 0 in nums[i + 1:]:
                        result += nums.index(0, i + 1) - (i + 1)
                    else:
                        result += len(nums[i + 1:])
                results.append(result)
                result = 0
        for i in results:
            result = max(i, result)
        return result
                   

class Solution1:
    def longestSubarray(self, nums: List[int]) -> int:
        queue = deque([])
        zeros = 0
        result = 0
        for num in nums:
            queue.append(num)
            if not num:
                zeros += 1
                while queue and zeros > 1:
                    if not queue.popleft():
                        zeros -= 1
                        break
            result = max(result, len(queue) - 1)
        return result